### PR TITLE
docs: fix irregularities and update language

### LINKED ISSUE #9347: Fix and refactor documentation irregularities
## Fix and refactor documentation irregularities

There are irregularities in the docs for various components. For example, in the template code for [Modal](https://react.carbondesignsystem.com/?path=/docs/components-modal--default) and [ComposedModal](https://react.carbondesignsystem.com/?path=/docs/components-composedmodal--default), `<Buttton>` elements are used instead of `<Button>`.

Although the example above highlighted one component there are other instances of the same issue.

Fixing these issues may create a better experience for users when interacting with documentation or the design system as a whole.

### Additional information

From the docs:
```js
<ComposedModal>
  ...
  <ModalFooter>
    <Button
      kind="secondary"
      onClick={() => { (Run some action...) setOpen(false); }}>
      Another button
    </Buttton>
    <Button
      kind="secondary"
      onClick={() => { (Run some action...) setOpen(false); }}>
      Secondary button
    </Buttton>
    <Button
      kind="primary"
      onClick={() => { (Run some action...) setOpen(false); }}>
      Priamry button
    </Buttton>
  </ModalFooter>
</ComposedModal>
```

### PR DESCRIPTION
**Pull request by adamalston**
Closes #9347

Fix irregularities and update language used in many documentation files.

#### Changelog

**New**

- Improved language so that docs can be translated easier by users.

**Changed**

- Fixed irregularities so that code templates were accurate.

**Removed**

- N/A

#### Testing / Reviewing

Check spelling/format of code blocks and comments in the documentation

### CODE REVIEW

**Comment by adamalston** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by adamalston** [comment]
recheck

**Review by sstrubberg — CHANGES_REQUESTED** [review]
This is incredible. Thank you for finding all these typos. Ready to approve once you get your merge conflict resolved! 🔥 

Re-request a review from me when you're ready so I get notified.

**Comment by adamalston** [comment]
Yea it's quite a lot. If you noticed in the files changed for this PR, all of them are under `packages/react`. I suspect when I dive into the other packages (in future PRs), there will be even more corrections that need to be made. 

Within my team, I know there is a translation push. Extrapolating a bit, spelling errors make text more difficult to translate so just fixing these inconsistencies in the documentation may give some users a better experience when interacting with the Carbon Design System.

**Review by tw15egan — APPROVED** [review]

File: packages/react/src/components/Slider/Slider.js [hunk 1]
```diff
@@ -367,7 +367,7 @@ export default class Slider extends PureComponent {
     // If shift was held, account for the stepMultiplier
     if (evt.shiftKey) {
       const stepMultiplier =
```
Comment (sstrubberg) [inline packages/react/src/components/Slider/Slider.js]: image

Looks like CircleCI is getting choked up around here.

**Review by sstrubberg — CHANGES_REQUESTED** [review]
You might also need to run `yarn lint` to clear up linting errors.

File: packages/react/src/components/Slider/Slider.js [hunk 2]
```diff
@@ -140,11 +140,11 @@ export default class Slider extends PureComponent {
 
     /**
      * A value determining how much the value should increase/decrease by Shift+arrow keys,
-     * which will be `(max - min) / stepMuliplier`.
+     * which will be `(max - min) / stepMultiplier`.
      */
-    stepMuliplier: deprecate(
+    stepMultiplier: deprecate(
```
Comment (jnm2377) [inline packages/react/src/components/Slider/Slider.js]: This should not change. Changing the prop name would be breaking, that's why it's been deprecated in favor of a different prop. Please revert these changes.

**Review by jnm2377 — CHANGES_REQUESTED** [review]

**Review by sstrubberg — APPROVED** [review]

**Review by jnm2377 — APPROVED** [review]

**Pull request merged**